addons/account/tests/test_tax.py

```diff
@@ -65,14 +67,82 @@ def _check_compute_all_results(self, taxes, expected_values, price_unit, **kwarg
+        evaluation_context_kwargs = evaluation_context_kwargs or {}
+        quantity = evaluation_context_kwargs.pop('quantity', 1)
+        compute_kwargs = compute_kwargs or {}
+        is_round_globally = evaluation_context_kwargs.get('rounding_method') == 'round_globally'
+        return {
+            'expected_values': expected_values,
+            'params': {
+                'test': 'taxes_computation',
+                'tax_values_list': taxes._convert_to_dict_for_taxes_computation(),
+                'price_unit': price_unit,
+                'quantity': quantity,
+                'evaluation_context_kwargs': evaluation_context_kwargs,
+                'compute_kwargs': compute_kwargs,
+                'is_round_globally': is_round_globally,
+            },
+        }
+
+    def _prepare_adapt_price_unit_to_another_taxes_test(self, price_unit, original_taxes, new_taxes, expected_price_unit):
+        return {
+            'expected_price_unit': expected_price_unit,
+            'params': {
+                'test': 'adapt_price_unit_to_another_taxes',
+                'original_tax_values_list': original_taxes._convert_to_dict_for_taxes_computation(),
+                'new_tax_values_list': new_taxes._convert_to_dict_for_taxes_computation(),
+                'price_unit': price_unit,
+            },
+        }
+
+    def _add_test_py_results(self, test):
+        params = test['params']
+        if params['test'] == 'taxes_computation':
+            evaluation_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
+                params['price_unit'],
+                params['quantity'],
+                **params['evaluation_context_kwargs'],
+            )
+            taxes_computation = self.env['account.tax']._prepare_taxes_computation(params['tax_values_list'], **params['compute_kwargs'])
+            test['py_results'] = {
+                'results': self.env['account.tax']._eval_taxes_computation(taxes_computation, evaluation_context),
+            }
+
+            if params['is_round_globally']:
+                evaluation_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
+                    test['py_results']['results']['total_excluded'] / params['quantity'],
+                    params['quantity'],
+                    **params['evaluation_context_kwargs'],
+                    reverse=True,
+                )
+                test['py_results']['reverse_results'] = self.env['account.tax']._eval_taxes_computation(taxes_computation, evaluation_context)
+        elif params['test'] == 'adapt_price_unit_to_another_taxes':
+            test['py_results'] = self.env['account.tax']._adapt_price_unit_to_another_taxes(
+                params['price_unit'],
+                params['original_tax_values_list'],
+                params['new_tax_values_list'],
+            )
```

Comment:
Protect against typo, and make it easier to detect that the issue if instead of only seeing that `py_result` is missing
```suggestion
            )
        else:
            assert False, f"Unknown tax test method: {params['test']}"
```